### ISSUE TITLE
Offline Version

### ISSUE THREAD

**Issue by bauerj** [bauerj/paperless_app#1]
For situations where you don't have internet access and need to access a document (e.g. you need to show a document at the Chinese border) offline access would be useful.

This includes the following tasks:

* [ ] Store list of documents locally
* [ ] Implement search locally
* [ ] Implement filtering locally
* [ ] Download all documents that are not downloaded yet
  * [ ] Make this behaviour configurable (default: off)
  * [ ] Also request all pages to make sure no documents are missing

**Comment by mvdkleijn** [comment]
I'd like to suggest that this feature could be limited to use only documents that have specified tag... "offline" or "local" comes to mind.

**Comment by Animizio** [comment]
I also would like something like encryption for the offline files. So the files are not free for everyone who's uses the smartphones file manager (this makes sense after the app has some security like fingerprint).

Also another thing, I don't want to open a new issue for it:
Could you add 2 different buttons after clicking a file, one for downloading a local copy (decrypted so ready for later use.. Like email.) and another to open the file to only read it. Not every "reader" supports to download the file to use it for something else. Thanks for your hard work!

**Comment by bauerj** [comment]
Hey,

the downloaded files should only be readable in the Paperless App, nowhere else. If this is different for you, something is broken here.

Hmm, what do you want to do with the file after downloading? Would it make sense to add a "Share" button instead?

**Comment by Animizio** [comment]
Yeah share would be okay too I think. So it's easier to use it with email apps etc.

**Comment by amenk** [comment]
For a first version maybe we do not need all the meta data and search locally - a simple list of offline documents might be sufficient for some cases. Currently I keep important documents in ownCloud as well for such cases.

**Comment by bauerj** [comment]
Yes, that would be a simple solution but due to server-side pagination and possible document changes we would need some more logic to make it happen:

- Request all documents (with tag offline), page by page
- Combine the responses, save them on disk
- For each document, check if PDf is missing or outdated

Not too complicated but it still needs some work.
